### PR TITLE
Front end scrolling

### PR DESCRIPTION
**Pull request by gwimm**
have fun everyone, will be awake in 6

### CODE REVIEW

File: rekuho/src/lib/Portion.svelte [hunk 1]
```diff
@@ -8,25 +8,29 @@
...
+  let width: pixels
+  let height: pixels
 
   const gutterWidth = 50 // TODO: maybe should be part of theme, minimum value?
   let input: HTMLTextAreaElement
   let container: HTMLElement
 
   const dispatch = createEventDispatcher<{
-    resize: [number, number]
+    resize: [pixels, pixels]
```
Comment (elkowar) [inline rekuho/src/lib/Portion.svelte:30]: For consistency, shouldn't we also make this an object with `width` and `height` or sth? Although I guess especially here with the pixels typealias this works 🤷🏻

File: rekuho/src/lib/config.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  font: {
+    family: "monospace",
+    weight: "normal",
+    size: "20px",
+  },
+
+  gutterWidth: 50,
+  scrollbarWidth: 12,
+  textOffset: 10,
+}
```
Comment (elkowar) [inline rekuho/src/lib/config.ts:35]: You really godda stop putting refactors in every commit, lmao
like sure, whatever, let's do it! But this really ought to be a separate commit and PR, ideally :(

File: rekuho/src/lib/event.ts [hunk 3]
```diff
@@ -38,3 +43,18 @@ export const keyboardToKeyInput = (event: KeyboardEvent): KeyInput | null => {
...
+      console.log(`wheel_event { delta: ${event.deltaY} }`)
+      break
+    case WheelEvent.DOM_DELTA_LINE:
+      console.error("unhandled page wheel line mode")
+      break
+    case WheelEvent.DOM_DELTA_PAGE:
+      console.error("unhandled page wheel delta mode")
+      break
+  }
+  return event.deltaY
```
Comment (elkowar) [inline rekuho/src/lib/event.ts:59]: why do we have this code twice, once here and once in Portion.svelte? this should just be a utility function or sth that's used in the other module, ideally

File: rekuho/src/lib/Portion.svelte [hunk 4]
```diff
@@ -59,20 +61,58 @@
...
+  // const scrollShowLine = (lineNumber: line, height: pixels): void => {
+  //   const lineOffset = lineNumber * lineHeight
+  //   if (lineOffset + lineHeight > scrollOffset + height) {
+  //     scrollOffset = lineOffset - height + lineHeight
+  //   } else if (scrollOffset > lineOffset) {
+  //     scrollOffset = lineOffset
+  //   }
+  // }
+
+  // $: scrollShowLine($state.carets[0]?.line ?? 0, height)
```
Comment (elkowar) [inline rekuho/src/lib/Portion.svelte:87]: why is there weird commented out code here? scrolling a line into view should be a backend responsibility, thus this shouldn't really ever be required in the frontend

**Review by elkowar — CHANGES_REQUESTED** [review]
Very nice! Just a few small style things ^^

**Comment by elkowar** [comment]
Well, and rustfmt, ofc

**Review by druskus20 — APPROVED** [review]
I trust @elkowar because I can't review from my device rn

File: rekuho/src/lib/Portion.svelte (hunk 1 above)
Comment (gwimm) [reply rekuho/src/lib/Portion.svelte:30]: [x, y]
